staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -404,10 +405,18 @@ func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 	err = <-errorChan
 	if err != nil {
 		runtime.HandleError(err)
-		pf.streamConn.Close()
+		if isFatalError(err) {
+			pf.streamConn.Close()
+		}
 	}
 }
 
+// isFatalError determines whether a given error should result in the closing of the listener connection
+func isFatalError(err error) bool {
+	return !net2.IsConnectionReset(err) &&
```

Comment:
the error that comes from `errorChan` it seems that is always wrapped

```go
	errorChan := make(chan error)
	go func() {
		message, err := ioutil.ReadAll(errorStream)
		switch {
		case err != nil:
			errorChan <- fmt.Errorf("error reading from error stream for port %d -> %d: %v", port.Local, port.Remote, err)
		case len(message) > 0:
			errorChan <- fmt.Errorf("an error occurred forwarding %d -> %d: %v", port.Local, port.Remote, string(message))
		}
		close(errorChan)
	}()
```

can this ` !net2.IsConnectionReset(`  happen?